### ISSUE TITLE
Repo URL not correct

### ISSUE THREAD

**Issue by nk2580** [afragen/github-updater#288]
When downloading from a private Bit Bucket Repo the URL for the update omits the tag reference for the latest version.

instead of:
/get/1.1.0.zip

the plugin looks for:
/get/ .zip

**Comment by afragen** [comment]
What branch and version are you using? I cannot replicate this.

**Comment by afragen** [comment]
Can you add `afragen` with `read` access to the specific repo so I can test?

**Comment by KevinNortier** [comment]
I had the same problem and i was looking for the where the issue was.

It seems to wok by adding a:
		if ( empty( $this->type->branch ) ) {
			$this->type->branch = 'master';
		}
to the construct_download_link function in the Bitbucket_API.php file. I added it before the other if-statements. 
I hope this is of help :)

https://github.com/afragen/github-updater/blob/develop/src/GitHub_Updater/Bitbucket_API.php#L266

**Comment by afragen** [comment]
@KevinNortier - That shouldn't be needed as an empty branch is set much before that.

https://github.com/afragen/github-updater/blob/develop/src/GitHub_Updater/Plugin.php#L164

**Comment by afragen** [comment]
Try the `develop` branch and see if this persists.

**Comment by nk2580** [comment]
Hi,

thanks everyone for the assistance.
I figured out that if you enable branch switching it defaults back to the master branch of the repo.

also tried the empty checker above which solved the problem when branch switching isn't enabled.

it might not be a bad idea to include the checker above in a later release for users not using branch switching.

(this is my first contribution to a GitHub project so if its a horrible idea, by all means don't consider it.)

**Comment by afragen** [comment]
@nk2580 thanks for the issue report. You're doing fine. :+1: 

The issue for me is that the line I referenced above, https://github.com/afragen/github-updater/blob/develop/src/GitHub_Updater/Plugin.php#L164 sets the branch to 'master' if the field is empty. This happens way before the code you're referencing.

Can you make sure you're using the current version of either branches and if you could give `afragen` temporary `read` access I could at least test with the repo that seems to be causing your problems. 

Thanks.

**Comment by KevinNortier** [comment]
I was checking around the Plugin.php row you mentioned and was looking for the same thing in Theme.php. I found that a similar check was missing, and saw that you've already added it to Theme.php
https://github.com/afragen/github-updater/blob/develop/src/GitHub_Updater/Theme.php#L156

I tried to just quickly test it by adding that row manually, and just wanted to say that it seems to have fixed the issue for me. Thanks!

(I guess i should have mentioned it was an issue with updating a theme and not a plugin...)
